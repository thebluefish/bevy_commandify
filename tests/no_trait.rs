use bevy::ecs::system::CommandQueue;
use bevy::prelude::*;
use bevy_commandify::*;

#[command(no_trait)]
fn foo(_world: &mut World) {}

#[test]
fn struct_command_still_works() {
    let mut world = World::new();

    let mut queue = CommandQueue::default();
    let mut commands = Commands::new(&mut queue, &mut world);

    commands.add(FooCommand);

    queue.apply(&mut world);
}
