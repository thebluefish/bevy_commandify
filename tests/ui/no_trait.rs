use bevy_commandify::*;
use bevy::prelude::*;
use bevy::ecs::system::CommandQueue;

#[command(no_trait)]
fn foo(world: &mut World) { }

/// Test that direct trait method fails
fn main() {
    let mut world = World::new();

    let mut queue = CommandQueue::default();
    let mut commands = Commands::new(&mut queue, &mut world);

    commands.foo();

    queue.apply(&mut world);
}