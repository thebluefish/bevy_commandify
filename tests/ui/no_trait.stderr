error[E0599]: no method named `foo` found for struct `bevy::prelude::Commands` in the current scope
  --> tests/ui/no_trait.rs:15:14
   |
15 |     commands.foo();
   |              ^^^ method not found in `Commands<'_, '_>`
